dbms/src/Storages/DeltaMerge/SegmentReadTask.h

```diff
@@ -113,21 +102,84 @@ struct SegmentReadTask
         RUNTIME_CHECK(input_stream != nullptr);
         return input_stream;
     }
+
+    GlobalSegmentID getGlobalSegmentID() const
+    {
+        return GlobalSegmentID{
+            .store_id = store_id,
+            .keyspace_id = dm_context->keyspace_id,
+            .physical_table_id = dm_context->physical_table_id,
+            .segment_id = segment->segmentId(),
+            .segment_epoch = segment->segmentEpoch(),
+        };
+    }
 };
 
+// Used in SegmentReadTaskScheduler, SegmentReadTaskPool.
+using MergingSegments = std::unordered_map<GlobalSegmentID, std::vector<UInt64>>;
+
 } // namespace DB::DM
 
 template <>
 struct fmt::formatter<DB::DM::SegmentReadTaskPtr>
 {
+    static constexpr auto parse(format_parse_context & ctx) { return ctx.begin(); }
+
     template <typename FormatContext>
-    auto format(const DB::DM::SegmentReadTaskPtr & t, FormatContext & ctx) const -> decltype(ctx.out())
+    auto format(const DB::DM::SegmentReadTaskPtr & t, FormatContext & ctx) const
     {
         return format_to(
             ctx.out(),
-            "{}_{}_{}",
+            "s{}_k{}_t{}_{}_{}_{}",
```

Comment:
```suggestion
            "s{}_ks{}_t{}_{}_{}_{}",
```
align with the naming in the physical directory name